Tests/base/GSFFIInvocation/basic.m

```diff
@@ -0,0 +1,115 @@
+#import <Foundation/Foundation.h>
+#import "Testing.h"
+#import "ObjectTesting.h"
+
+@protocol NSMenuItem <NSObject>
+- (NSString*) keyEquivalent;
+- (void) setKeyEquivalent: (NSString*)aKeyEquivalent;
+@end
+
+@interface NSMenuItem : NSObject <NSMenuItem>
+{
+  NSString *_keyEquivalent;
+}
+@end
+
+@implementation NSMenuItem
+- (void) setKeyEquivalent: (NSString*)aKeyEquivalent
+{
+  ASSIGNCOPY(_keyEquivalent,  aKeyEquivalent);
+}
+
+- (NSString*) keyEquivalent
+{
+  return _keyEquivalent;
+}
+@end
+
+@interface GSFakeNSMenuItem : NSObject
+{
+  NSMenuItem* _originalItem;
+}
+
+- (id) initWithItem: (NSMenuItem*)item;
+- (NSMenuItem*) originalItem;
+- (id) target;
+- (SEL)action;
+- (void) action: (id)sender;
+@end
+
+@implementation GSFakeNSMenuItem
+- (id) initWithItem: (NSMenuItem*)item
+{
+  self = [super init];
+  if (self)
+  {
+    _originalItem = item;
+  }
+  return self;
+}
+
+- (NSMenuItem*) originalItem
+{
+  return _originalItem;
+}
+
+- (id)target
+{
+  return self;
+}
+
+- (SEL)action
+{
+  return @selector(action:);
+}
+
+- (id)forwardingTargetForSelector:(SEL)selector
+{
+  if ([_originalItem respondsToSelector:selector])
+    return _originalItem;
+  return nil;
+}
+
+- (void)forwardInvocation:(NSInvocation *)invocation
+{
+  SEL selector = [invocation selector];
+
+  // Forward any invocation to the original item if it supports it...
+  if ([_originalItem respondsToSelector:selector])
+    [invocation invokeWithTarget:_originalItem];
+}
```

Comment:
In the else case you should call the super implementation, to make sure we end up in -doesNotRecognizeSelector:. This won't change the problem, just document the correct way to implement this method.